### DISCUSSION TITLE
How to add "nowplaying" song in "channel topic"

### DISCUSSION THREAD

**Discussion by Vato9394** [jagrosh/MusicBot#801] [Questions] [answered]
On the setup page you see this gif:
![68747470733a2f2f692e696d6775722e636f6d2f6b5674544b76532e676966](https://user-images.githubusercontent.com/24278426/133419196-418a1b7b-36a3-471f-9de7-26b7a81f1788.gif)

there, you can see that the currently playing song is integrated in the channel topic as such:
![currently playing in channel topic](https://user-images.githubusercontent.com/24278426/133419086-a6513574-d255-4ffe-ac0d-74046588d857.png)

Can someone please help me on how to add this into the channel topic?
(and maybe update the readme?)

thanks!

**Comment by MichailiK** [comment]
Your bot requires Manage Channels permission to update the channel topic.

Additionally, Discord has changed channel topic rate limits since then: You can now only change the channel topic 2 times every 15 minutes.

**Comment by Vato9394** [comment]
I understand the 15 minute limit. But can you tell me on how to integrate this?
(lets say I only put mixes of 1 hour in the bot)
I already gave the bot "Manage Channel Permission" but this still does not work.

**Reply by MichailiK** [reply]
It should already work out of the box if your bot has Manage Channels.

**Comment by jagrosh** [comment, marked as answer]
The bot will update a channel's topic if 2 things are the case:
1. You have used the `settc` command to limit commands to that text channel
2. The bot has the Manage Channel permission in that same text channel

As already stated, due to Discord changing ratelimits since that feature was introduced, the topic can only change a maximum of 2 times per 15 minutes.

**Comment by Vato9394** [comment]
Nice, it is working now, thanks!
